pkg/spoke/controllers/deletioncontroller/deletion_controller.go

```diff
@@ -66,10 +73,16 @@ func (m *StaleManifestDeletionController) sync(ctx context.Context, controllerCo
 		return err
 	}
 
-	return m.syncManifestWork(ctx, manifestWork)
+	return m.syncManifestWork(ctx, manifestWork, func() {
+		controllerContext.Queue().AddAfter(manifestWorkName, m.rateLimiter.When(manifestWorkName))
+	}, func() {
+		m.rateLimiter.Forget(manifestWorkName)
+	})
 }
 
-func (m *StaleManifestDeletionController) syncManifestWork(ctx context.Context, originalManifestWork *workapiv1.ManifestWork) error {
+// syncManifestWork collects latest applied resources from ManifestWork.Status.ResourceStatus and merges them with
+// existing applied resources. It also deletes applied resources for stale manifests
+func (m *StaleManifestDeletionController) syncManifestWork(ctx context.Context, originalManifestWork *workapiv1.ManifestWork, requeueWork requeueWorkFunc, forgetWork forgetWorkFunc) error {
```

Comment:
passing in requeueWork and forgetWork seems unnecessary since it could exist on `m`.  I'm guessing you've set this up for unit testing, so I don't object to the idea of stubs, but I would suggest something like

```go
type StaleManifestDeletionController struct{
  // stuff
  requeueWorkUnitTestFn func()
  forgetWorkUnitTestFn func()
}

func (m *Stale) requeueWork(){
  if m.requeueWorkUnitTestFn != nil{
    // unit testing
    m.requeueWorkUnitTestFn()
    return
  }

  controllerContext.Queue().AddAfter(manifestWorkName, m.rateLimiter.When(manifestWorkName))
}

```